lib/my_app.rb

```diff
@@ -0,0 +1,151 @@
+ENV['RACK_ENV'] ||= "development"
+
+require_relative '../data_mapper_setup.rb'
+require 'sinatra/base'
+require 'sinatra/flash'
+
+class MakersBnb < Sinatra::Base
+  use Rack::MethodOverride
+  register Sinatra::Flash
+
+  enable :sessions
+  set :session_secret, 'super secret'
+
+  helpers do
+    def current_user
+      @current_user ||= User.get(session[:user_id]) 
+    end
+  end
+
+  get '/' do
+    @spaces = Space.all
+    erb :index
+  end
+
+  post '/' do
+    a_space = Space.create(params)  
+    current_user.space << a_space
+    current_user.save
+    redirect to('/new')
+  end
+
+  get '/new' do 
+    erb :newPost
+  end
+
+  get '/users/new' do
+    @user = User.new
+    erb :new_user
+  end
+
+  post '/users' do
+    @user = User.create(email: params[:email],
+                        password: params[:password],
+                        password_confirmation: params[:password_confirmation],
+                        user_name: params[:user_name])
+    if @user.save
+      session[:user_id] = @user.id
+      redirect to '/'
+    else
+      flash.now[:errors] = @user.errors.full_messages
+      erb :new_user
+    end
+  end
+
+  get '/sessions/new' do
+    erb :'sessions/new'
+  end
+
+  get '/booked/:id' do 
+    space = Space.last(id: params[:id])
+    if current_user != nil
+      owner = space.user.last.user_name
+      user = current_user.user_name
+      @booking_object = Booking.create(name: user, owner: owner, room: space.name, confirmation: params[:false])
+      session[:booking_ex] = @booking_object.id
+      @booking = Booking.last(id: session[:booking_ex])
+    end
+      erb :confirmation
+  end
+
+  get '/booking-requests' do 
+    if current_user
+      @bookings = Booking.all(owner: current_user.user_name)
```

Comment:
This is worrying me - use something that can't be duplicated as the relation key. What if two people have the same name? I can see all their booking requests?

What database column cannot be duplicated, and auto-increments to avoid this kind of conflict?

Secondly, I don't like that `Booking` is owned by an `owner`, not a `user`. Inconsistent naming will rapidly undo all your domain modelling.